src/hotspot/share/opto/type.hpp

```diff
@@ -157,7 +157,7 @@ class Type {
   static const TypeInfo _type_info[];
 
   static int uhash( const Type *const t );
-  // Structural equality check.  Assumes that cmp() has already compared
+  // Structural equality check.  Assumes that equals() has already compared
```

Comment:
Is this even correct? Because we use `eq` inside `equals`, (before your patch, we used `eq` inside `cmp`).
Should this not rather mean that we should have already done the `==`?
What do you think?

Reply:
I think it's still correct, because from my understanding it's referring to this line in `equals`:
```c++
if (t1->_base != t2->_base) {
  return false; // Missed badly
}
```
`equals` only calls `eq`after this, so `eq` knows that the underlying type of both classes are the same. So `TypeInt::eq`, for example, is able to safely cast the incoming `Type* t` to a `TypeInt*` to do a structural equality check.